pkg/eks/api.go

```diff
@@ -257,6 +257,14 @@ func (c *ClusterProvider) EnsureAMI(version string, ng *api.NodeGroup) error {
 		return ami.NewErrNotFound(ng.AMI)
 	}
 
+	// Get Root Device Name
+	rootDevice, err := ami.LookupRootDeviceName(c.Provider.EC2(), ng.AMI)
+	if err != nil {
+		return errors.Wrapf(err, "%s failed root device name lookup", ng.AMI)
+	}
+
+	ng.VolumeName = rootDevice
```

Comment:
Actually, one should be able to override this by setting `volumeName` (if they must). So please make this lookup optional. You'd want to use `*string` for the field type instead of `string`, and use `if api.IsSetAndNonEmptyString(ng.VolumeName)` here.

Reply:
Also, we will need validation of the combination of all 3 fields being set in a sane way, take a look at `pkg/apis/eksctl.io/v1alpha5/validation.go`, e.g. `volumeName` or `volumeType` cannot be set without `volumeSize`, but `volumeSize` alone is okay. The preference is to use pointers for all fields, so do feel free to convert `volumeSize` to `*int` and `volumeType` to `*string` for consistency.

Reply:
> Actually, one should be able to override this by setting `volumeName` (if they must). So please make this lookup optional. You'd want to use `*string` for the field type instead of `string`, and use `if api.IsSetAndNonEmptyString(ng.VolumeName)` here.

I am not sure why you would want to override something that is being looked up. Is that so you can add an extra disk to the node instead of increasing the size of the root volume?

Reply:
> I am not sure why you would want to override something that is being looked up.

So that you can avoid the look-up. Some users prefer to avoid extra APIs calls due to access restrictions. I don't remember, do we make the `DescribeImages` call unconditionally when we call `IsAvailable`, or there is a way to avoid that right now? If we call it unconditionally now, then don't worry too much.